drivers/adc/adc_stm32.c

```diff
@@ -807,13 +830,29 @@ static void dma_callback(const struct device *dev, void *user_data,
 			 * within adc_context_start_sampling
 			 */
 			dma_stop(data->dma.dma_dev, data->dma.channel);
+
+			if (data->ctx.options.batch_mode) {
+				/* In batch mode the ADC must be stopped manually
+				 * because it's running in continuous mode.
+				 */
+				LL_ADC_REG_StopConversion(adc);
```

Comment:
Shouldn't we just extract the stopping sequence from adc_stm32_disable to a separate function and call it here?